py/server/deephaven/table.py

```diff
@@ -319,6 +320,34 @@ def meta_table(self) -> Table:
     def j_object(self) -> jpy.JType:
         return self.j_table
 
+    def get_table_attributes(self) -> Dict[str, Any]:
+        """Returns all the attributes defined on the table."""
+        j_map = jpy.cast(self.j_table, _JLiveAttributeMap).getAttributes()
+        return j_map_to_dict(j_map)
+
+    def with_table_attributes(self, attrs: Dict[str, Any]) -> Table:
+        """Returns a new Table that has the provided attributes defined on it and shares the underlying data and schema
+        with this table.
+
+        Note, the table attributes are immutable once defined, and are mostly used internally by the Deephaven
+        engine. For advanced users, certain predefined plug-in attributes provide a way to extend Deephaven with
+        custom-built plug-ins.
+
+        Args:
+            attrs (Dict[str, Any]): a dict of table attribute names and their values
+
+        Returns:
+            a new Table
+
+        Raises:
+            DHError
+        """
+        try:
+            j_map = j_hashmap(attrs)
```

Comment:
Are we unwrapping objects as they're copied into the map? Should we?

Reply:
Yes and we should, otherwise we'll run into a problem with JPY when the value is a wrapped Java object.